pkg/clients/etcd.go

```diff
@@ -0,0 +1,40 @@
+package clients
+
+import (
+	"fmt"
+	"time"
+
+	logging "github.com/op/go-logging"
+
+	"github.com/coreos/etcd/client"
+)
+
+type EtcdConfig struct {
```

Comment:
> To me, it looks like a lot of the ABP code is in their own files. and client.go should all be in the new package. I am wondering what methods you were thinking should be left in apb?

I don't think code that's responsible for the execution of an apb belongs in the `clients` pkg, it's a natural fit for the `apb` pkg. It's effectively `ExecuteApb(method, parameters)`, and we should hide the guts of **how** that actually gets done behind that function (in this case, it's running the pod). `ExecuteApb` IMO is a better name for `RunImage`. Run image bleeds the abstraction with "Image" in the name. It suggests "run this apb method with these parameters, I don't care how you do that" vs "run this image because that's how we happen to run apbs".